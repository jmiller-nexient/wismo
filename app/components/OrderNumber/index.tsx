import * as React from 'react';

interface IOrderNumberProps {
  value: string;
}

const OrderNumber: React.FC<IOrderNumberProps> = (props: IOrderNumberProps): React.ReactElement => {
  const { value } = props;

  const formatOrderNumber = (input: string): string => {
    return `${input.substr(0,4)} ${input.substr(4,4)} ${input.substr(8,3)} ${input.substr(11,1)}`;
  };

  if (value.match(new RegExp('([0-9]{12})\d', 'g'))) {
    throw new Error('order numbers must be exactly 12 digits.');
  }

  return (
    <span>
      {formatOrderNumber(value)}
    </span>
  );
};

export default OrderNumber;
