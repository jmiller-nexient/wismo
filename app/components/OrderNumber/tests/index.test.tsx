import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';

import OrderNumber from '..';
import {shallow} from 'enzyme';

describe('<OrderNumber />', () => {
  it('should render correctly when given a valid value', () => {
    const tree: ReactElement = renderer.create(
      <OrderNumber value={'112233445566'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should throw an error when order number has non-numeric characters', () => {
    expect(() =>  shallow(<OrderNumber value="ABCDEFGHIJKL" />)).toThrowError();
  });

  it('should throw an error when order number is less than 12 digits', () => {
    expect(() => shallow(<OrderNumber value={'123456'}/>)).toThrowError();
  });

  it('should throw an error when order number is more than 12 digits', () => {
    expect(() => shallow(<OrderNumber value={'1234561234567'}/>)).toThrowError();
  });
});
