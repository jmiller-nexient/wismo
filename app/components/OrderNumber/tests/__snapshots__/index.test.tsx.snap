// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderNumber /> should render correctly when given a valid value 1`] = `
<span>
  1122 3344 556 6
</span>
`;
